ledger/acctupdates_test.go

```diff
@@ -1030,3 +1030,92 @@ func TestListCreatables(t *testing.T) {
 		expectedDbImage, numElementsPerSegement)
 	listAndCompareComb(t, au, expectedDbImage)
 }
+
+func TestIsWritingCatchpointFile(t *testing.T) {
+
+	au := &accountUpdates{}
+
+	au.catchpointWriting = make(chan struct{}, 1)
+	ans := au.IsWritingCatchpointFile()
+	require.True(t, ans)
+
+	close(au.catchpointWriting)
+	ans = au.IsWritingCatchpointFile()
+	require.False(t, ans)
+}
+
+func TestGetCatchpointStream(t *testing.T) {
+
+	proto := config.Consensus[protocol.ConsensusCurrentVersion]
+
+	ml := makeMockLedgerForTracker(t, true)
+	defer ml.close()
+	ml.blocks = randomInitChain(protocol.ConsensusCurrentVersion, 10)
+
+	accts := []map[basics.Address]basics.AccountData{randomAccounts(20, true)}
+	au := &accountUpdates{}
+	conf := config.GetDefaultLocal()
+	conf.CatchpointInterval = 1
+	au.initialize(conf, ".", proto, accts[0])
+	defer au.close()
+
+	err := au.loadFromDisk(ml)
+	require.NoError(t, err)
+
+	filesToCreate := 4
+
+	// Prepare the directory as GetCatchpointStream expects
+	err = os.Mkdir("catchpoints", 0777)
+	require.NoError(t, err)
+
+	// Create the catchpoint files with dummy data
+	for i := 0; i < filesToCreate; i++ {
+		f, err := os.Create(fmt.Sprintf("catchpoints/%d.catchpoint", i))
```

Comment:
I think that `ioutil.WriteFile` could help you do this section much simpler, but there is nothing incorrect here.